src/fsharp/NameResolution.fs

```diff
@@ -3080,13 +3079,13 @@ let rec ResolvePartialLongIdentInModuleOrNamespace (ncenv: NameResolver) nenv is
     
     let tycons = 
         mty.TypeDefinitions
-        |> List.filter (fun tcref -> not (tcref.LogicalName.Contains(",")))
-        |> List.filter (fun tycon -> not (IsTyconUnseen ad g ncenv.amap m (modref.NestedTyconRef tycon)))
+        |> List.filter (fun tcref -> 
+            (not (tcref.LogicalName.Contains(","))) &&
```

Comment:
No parentheses needed, so ``not A && not B`` is fine rather than ``(not A) && (not B)``